tests/test_server.py

```diff
@@ -17,117 +41,284 @@ def get_app(self):
+        join_result, start_result = yield [
+            self.client.fetch(
+                self.get_url('/games/{}/board?token={}&alias={}'.format(
+                    ID, TOKEN, 'tester'
+                )),
+                method='GET'
+            ),
+            self.client.fetch(
+                self.get_url('/games/{}'.format(ID)),
+                method='POST',
+                body=json.dumps({'token': TOKEN})
+            )
+        ]
 
-    def test_join(self):
-        def callback(response):
-            response = response.body.decode()
+        self.assertEqual(join_result.code, 200)
+        self.assertEqual(start_result.code, 200)
 
-            self.assertIn("id: '{}'".format(self.user.id), response)
-            self.assertIn("name: '{}'".format(self.user.name), response)
-            self.stop()
+        with self.assertRaises(tornado.httpclient.HTTPError):
+            join_again = yield self.client.fetch(
+                self.get_url('/games/{}/board?token={}'.format(ID, TOKEN)),
+                method='GET',
+            )
+            self.assertEqual(join_again.code, 404)
+
+        game = json.loads(join_result.body.decode())
+        expected_game = {
+            'score': 0,
+            'moves': 5,
+            'moved': None,
+        }
+
+        for key, value in expected_game.items():
+            self.assertTrue(key in game)
+            self.assertEqual(game[key], value)
+
+        move = yield self.client.fetch(
+            self.get_url('/games/{}/board'.format(ID)),
+            method='POST',
+            body=json.dumps({'x': 3, 'y': 1, 'token': TOKEN}),
+        )
+        move_result = json.loads(move.body.decode())
+        self.assertTrue('score' in move_result)
+
+        round_result = yield self.client.fetch(
+            self.get_url('/games/{}'.format(ID)),
+            method='GET',
+            headers={
+                'Accept': 'html'
+            }
+        )
+        self.assertTrue(round_result.code, 200)
+
+        round_result_page = round_result.body.decode()
 
-        self.http_client.fetch(
-            self.get_url('/games/2/board?token={}'.format(self.user.token))
+        self.assertTrue("id: '{}'".format(ID) in round_result_page)
+        self.assertTrue(
+            "score: '{}'".format(move_result['score']) in round_result_page
         )
 
-        self.io_loop.call_later(
-            1,
-            lambda: self.http_client.fetch(
-                self.get_url('/games/2'),
-                callback,
+        deleted = yield self.client.fetch(
+            self.get_url('/games/{}?token={}'.format(ID, TOKEN)),
+            method='DELETE',
+        )
+        self.assertEqual(deleted.code, 200)
+        self.assertEqual(len(server.game_rooms), 0)
+
+    @unittest.mock.patch(
+        'server.game_rooms', {
+            ID: GameRoom(
+                _id=ID,
+                author=LOGIN
             )
+        }
+    )
+    @tornado.testing.gen_test(timeout=100000)
```

Comment:
I added `timeout=...` with huge value, to work with pdb (I could debug for longer then 10 seconds - default value). This is not necessary in now - please remove it from all @tornado.testing.gen_test decorators.